mailbox/backup/src/main/java/org/apache/james/mailbox/backup/zip/package-info.java

```diff
@@ -18,22 +18,33 @@
  ****************************************************************/
 
 /**
- * The zip archive format for the backup of an account
- * This archive contains the following entries for each mailbox of the user:
+ *  The zip archive format for the backup of an account
+ *
+ *  This archive contains the mailboxes of the account. And for each mailbox :
+ *   - its annotations
+ *   - its messages
+ *
+ *  This structure is repeated sequentially for each mailbox.
+ *  The order presented here is the order in which the entries are added to the zip file.
+ *  This is important and must be respected because during the restoration, the file is read iteratively in memory from an inputstream,
+ *  without storing the whole structure on memory nor on disk.
+ *
+ *  So we have by order of insertion mailbox1 followed by it's elements then mailbox2 and its elements ...
```

Comment:
we should change that in a next PR: it's a non sense. Either we store messages in mailboxes if they are related to them, or we store all messages flat in the archive.
I suggest that in a next PR we change that to allow each message to have several mailboxIds, and to store all the mailboxes (and their annotations), and then we store all the messages (only once per message).

Reply:
but thank you for documenting this weird implementation!